php/post-types/class-block-post.php

```diff
@@ -895,6 +900,15 @@ public function save_block( $data ) {
 		check_admin_referer( 'block_lab_save_fields', 'block_lab_fields_nonce' );
 		check_admin_referer( 'block_lab_save_properties', 'block_lab_properties_nonce' );
 
+		// Update post meta with allowed post types.
+		if ( isset( $_POST['block-post-types'] ) ) {
+			$post_types = sanitize_text_field(
+				wp_unslash( $_POST['block-post-types'] )
+			);
+			$post_types = explode( ',', $post_types );
+			update_post_meta( $post_id, 'block-post-types', $post_types );
```

Comment:
@kienstra I chose here to use post meta to store which post types a block should appear in (or not).

The alternative is to extend the `Block` class with a new property for post types.

Which do you think is a better approach?

Reply:
I think I just answered my own question. It's got to be an extension of Block, so that Export / Import continues to work properly.

Reply:
In addition, rather than saving "allowed post types", we might need to save "disallowed post types".

Use case:
- I create a new block, and don't change any "Post Type" settings, so that the new block works in all post types
- The block saves, storing all current post types as allowed
- I create a new post type, and attempt to add my block
- The block doesn't exist for this post type, since it hasn't been explicitly set
- In order to get my blocks to show for the new post type, I need to edit every block and remove the post type